cockroachdb/sqlalchemy/dialect.py

```diff
@@ -36,6 +37,21 @@
 )
 
 
+class _SavepointState(threading.local):
+    """Hack to override names used in savepoint statements.
+
+    To get the Session to do the right thing with transaction retries,
+    we use the begin_nested() method, which executes a savepoint. We
+    need to transform the savepoint statements that are a part of this
+    retry loop, while leaving other savepoints alone. Unfortunately
+    the interface leaves us with no way to pass this information along
+    except via a thread-local variable.
+    """
+    def __init__(self):
+        self.cockroach_restart = False
+savepoint_state = _SavepointState()
```

Comment:
That's just python's weird thread-local interface. Define a class (to give attributes default values in each new thread), then make one global instance of it. A blank line would probably help here.